pkg/discord/main.go

```diff
@@ -206,122 +207,146 @@ func New(dbConnection *gorm.DB) (*discordgo.Session, error) {
-		Keyword:    "onboard",
-		Handler:    onboard,
+		Keyword:       "onboard",
+		Handler:       onboard,
 		PermissionMap: permissions,
-		MinArgs: 0,
-		MaxArgs: 0,
+		MinArgs:       0,
+		MaxArgs:       0,
 	}
 	cmdHandler.RegisterCommand(onboardCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[0] = PermissionMember
 	onboardAllCommand := Command{
-		Keyword:    "onboardall",
-		Handler:    onboardAll,
+		Keyword:       "onboardall",
+		Handler:       onboardAll,
 		PermissionMap: permissions,
-		MinArgs: 0,
-		MaxArgs: 0,
+		MinArgs:       0,
+		MaxArgs:       0,
 	}
 	cmdHandler.RegisterCommand(onboardAllCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[0] = PermissionChannel
 	getAgendaCommand := Command{
-		Keyword:    "agenda",
-		Handler:    getAgenda,
+		Keyword:       "agenda",
+		Handler:       getAgenda,
 		PermissionMap: permissions,
-		MinArgs: 0,
-		MaxArgs: 0,
+		MinArgs:       0,
+		MaxArgs:       0,
 	}
 	cmdHandler.RegisterCommand(getAgendaCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[1] = PermissionMember
 	joinCommand := Command{
-		Keyword:    "join",
-		Handler:    joinProject,
+		Keyword:       "join",
+		Handler:       joinProject,
 		PermissionMap: permissions,
-		MinArgs: 1,
-		MaxArgs: 1,
+		MinArgs:       1,
+		MaxArgs:       1,
 	}
 	cmdHandler.RegisterCommand(joinCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[1] = PermissionMember
 	leaveCommand := Command{
-		Keyword:    "leave",
-		Handler:    leaveProject,
+		Keyword:       "leave",
+		Handler:       leaveProject,
 		PermissionMap: permissions,
-		MinArgs: 1,
-		MaxArgs: 1,
+		MinArgs:       1,
+		MaxArgs:       1,
 	}
 	cmdHandler.RegisterCommand(leaveCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[2] = PermissionAdmin
 	trackCommand := Command{
-		Keyword:    "track",
-		Handler:    trackFile,
+		Keyword:       "track",
+		Handler:       trackFile,
 		PermissionMap: permissions,
-		MinArgs: 2,
-		MaxArgs: 2,
+		MinArgs:       2,
+		MaxArgs:       2,
 	}
 	cmdHandler.RegisterCommand(trackCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[1] = PermissionAdmin
 	untrackCommand := Command{
-		Keyword:    "untrack",
-		Handler:    untrackFile,
+		Keyword:       "untrack",
+		Handler:       untrackFile,
 		PermissionMap: permissions,
-		MinArgs: 1,
-		MaxArgs: 1,
+		MinArgs:       1,
+		MaxArgs:       1,
 	}
 	cmdHandler.RegisterCommand(untrackCommand)
 
 	permissions = make(map[int]Permission)
 	permissions[1] = PermissionMember
 	fetchFileCommand := Command{
-		Keyword:    "fetch",
-		Handler:    fetchFileDispatch,
+		Keyword:       "fetch",
+		Handler:       fetchFileDispatch,
 		PermissionMap: permissions,
-		MinArgs: 1,
-		MaxArgs: 1,
+		MinArgs:       1,
+		MaxArgs:       1,
 	}
 	cmdHandler.RegisterCommand(fetchFileCommand)
 
+	permissions = make(map[int]Permission)
+	permissions[0] = PermissionMember
+	permissions[-1] = PermissionMember
```

Comment:
What is the purpose of assigning PermissionMember to both [0] index and [-1] index

Reply:
This did feel clunky. The way I set up permissions based on argument number requires each argument number to have it defined. Gonna rewrite that again, with the option of a simple permission to be set(the way it was before), and have the option to specify a function to determine what permission to use, like we had discussed on Monday.

Reply:
That would for sure make the code more readable/understandable.

Reply:
Updated